sled-agent/src/services.rs

```diff
@@ -918,10 +921,21 @@ impl ServiceManager {
                 underlay_address,
                 rack_id,
                 rack_network_config,
+                metrics_queue: metrics_queue.clone(),
             })
             .map_err(|_| "already set".to_string())
             .expect("Sled Agent should only start once");
 
+        // At this point, we've already started up the switch zone, but the
+        // VNICs inside it cannot have been tracked by the sled-agent's metrics
+        // task (the sled-agent didn't exist at the time we started the switch
+        // zone!). Notify that task about the zone's VNICs now.
+        if let SwitchZoneState::Running { zone, .. } =
+            &*self.inner.switch_zone.lock().await
+        {
+            metrics_queue.track_zone_links(zone).await;
```

Comment:
Just to confirm, as we're locking the switch zone here, the underlying opte/vnic sends to the metric queue drop errors on the floor, right? Should we log the errors at least or have a metric on the queue itself of tracked/untracked?

Reply:
Yeah, the call to `track_zone_links()` does not return errors, because an error is only returned if the channel is closed or the receiver dropped, neither of which is actionable. I'll look into logging the errors, that might be useful.

A metric on the queues would need to go to the `metrics_task` we're using to produce data. If we cannot send a message here, then the channel is closed anyway, so an attempt to send new data to it won't work either, I think.

Reply:
Added logging to call-sites in f9d4b5c3f